table/queries/derived_satellite_scans.sql

```diff
@@ -0,0 +1,205 @@
+                                   rcode INTEGER,
+                                   answers ANY TYPE) AS (
+    CASE 
+        WHEN dns_error IS NOT NULL THEN ClassifySatelliteError(dns_error)
+        WHEN rcode = -1 THEN ClassifySatelliteErrorNegRCode(dns_error)
+        WHEN rcode != 0 THEN ClassifySatelliteRCode(rcode)
+        WHEN ARRAY_LENGTH(answers) = 0 THEN "❗️answer:no_answer"
+        ELSE IFNULL(
+            (SELECT InvalidIpType(answer.ip) FROM UNNEST(answers) answer LIMIT 1),
+            CASE
+                WHEN (SELECT LOGICAL_OR(answer.matches_control.ip)
+                      FROM UNNEST(answers) answer) THEN "✅answer:matches_ip"
+                WHEN (SELECT LOGICAL_OR(answer.http_analysis_is_known_blockpage)
+                      FROM UNNEST(answers) answer) THEN CONCAT("❗️page:http_blockpage:", answers[OFFSET(0)].http_analysis_page_signature)
+                WHEN (SELECT LOGICAL_OR(answer.https_analysis_is_known_blockpage)
+                      FROM UNNEST(answers) answer) THEN CONCAT("❗️page:https_blockpage:", answers[OFFSET(0)].https_analysis_page_signature)
+                WHEN (SELECT LOGICAL_OR(IsCertForDomain(a.https_tls_cert, domain_name))
+                      FROM UNNEST(answers) a) THEN "✅answer:cert_for_domain"
+                WHEN (SELECT LOGICAL_AND(NOT IsCertForDomain(a.https_tls_cert, domain_name))
+                      FROM UNNEST(answers) a) THEN CONCAT("❗️answer:cert_not_for_domain:", AnswersSignature(answers))
+                -- We check AS after cert because we've seen (rare) cases of blockpages hosted on the ISP that also hosts Akamai servers.
+                WHEN (SELECT LOGICAL_OR(answer.matches_control.asn)
+                      FROM UNNEST(answers) answer) THEN "✅answer:matches_asn"
+                ELSE CONCAT("❗️answer:not_validated:", AnswersSignature(answers))
+            END
+        )
+    END
+);
+
+
+# BASE_DATASET and DERIVED_DATASET are reserved dataset placeholder names
+# which will be replaced when running the query
+
+# Increment the version of this table if you change the table in a backwards-incomatible way.
+
+# Rely on the table name firehook-censoredplanet.derived.merged_reduced_scans_vN
+# if you would like to see a clear breakage when there's a backwards-incompatible change.
+# Old table versions will be deleted.
+CREATE OR REPLACE TABLE `firehook-censoredplanet.DERIVED_DATASET.reduced_satellite_scans_v1`
+PARTITION BY date
+# Column `country_name` is always used for filtering and must come first.
+# `network`, `subnetwork`, and `domain` are useful for filtering and grouping.
+CLUSTER BY country_name, network, subnetwork, domain
+OPTIONS (
+  friendly_name="Reduced Satellite Scans",
+  description="Filtered and pre-aggregated table of Satellite scans to use with the Censored Planed Dashboard"
+)
+AS (
+WITH Grouped AS (
+    SELECT
+        date,
+
+        IF(resolver_name="special","special",NET.REG_DOMAIN(resolver_name)) as reg_hostname,
```

Comment:
Honestly I don't know. All it says in the documentation is some resolvers are named `special` instead of having a hostname. https://docs.censoredplanet.org/dns.html#id2 The special case is added here because `NET.REG_DOMAIN` can't handle that string.